rclpy/rclpy/executors.py

```diff
@@ -762,7 +762,11 @@ def _spin_once_impl(
         except ConditionReachedException:
             pass
         else:
-            self._executor.submit(handler)
+            def handler_wrapper(handler):
```

Comment:
create object to raise the exception if there is, instead of managing the `future.result()` via `submit`.